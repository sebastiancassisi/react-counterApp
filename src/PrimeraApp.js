import React, { useState } from 'react';
import './index.css';
import { PropTypes } from 'prop-types';

const PrimeraApp = ({ miSaludo, subtitulo }) => {
  const saludo = 'Hola mundo !!';

  return (
    <>
      <h1>{saludo}</h1>
      <h2>{miSaludo}</h2>
      <h3>{subtitulo}</h3>
    </>
  );
};

PrimeraApp.propTypes = {
  miSaludo: PropTypes.string,
};

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo',
};

export default PrimeraApp;
