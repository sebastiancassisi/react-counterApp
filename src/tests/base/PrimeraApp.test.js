import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import PrimeraApp from '../../PrimeraApp';
import React from 'react';
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp/>', () => {
  /*   test('debe de mostar el mensaje "Hola, Soy sebastian"', () => {
    const saludo = 'Hola, Soy sebastian';

    const { getByText } = render(<PrimeraApp saludo={saludo} />);

    expect(getByText(saludo)).toBeInTheDocument();
  }); */

  test('debe de mostar <PrimeraApp/> correctamente"', () => {
    const saludo = 'Hola, Soy sebastian';
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostar el subtitulo enviado por props', () => {
    const saludo = 'Hola, Soy sebastian';
    const subtitulo = 'Soy un subtitulo';
    const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);

    const textoParrafo = wrapper.find('h3').text();

    expect(textoParrafo).toBe(subtitulo);
  });
});
