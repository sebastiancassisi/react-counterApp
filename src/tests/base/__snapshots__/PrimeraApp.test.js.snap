// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mostar <PrimeraApp/> correctamente" 1`] = `
<Fragment>
  <h1>
    Hola mundo !!
  </h1>
  <h2 />
  <h3>
    Soy un subtitulo
  </h3>
</Fragment>
`;
